### PR TITLE
add jupyter-book team as maintainers

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR adds @conda-forge/jupyter-book team to the maintainers to help keep releases flowing and figure out multi-feedstock issues.

### CODE REVIEW

**Comment by mmcky** [comment]
sorry @bollwyvl I didn't see this. Should I add the `conda-forge/jupyter-book` team to the list of maintainers?

**Comment by bollwyvl** [comment]
Sure! Just here to help (keep jupyter-book working)!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1,2 @@
+* @conda-forge/jupyter-book
```
[the remark this answers is not in the document]
Comment (bollwyvl) [reply .github/CODEOWNERS:1]: Yes, it's an existing team: note that it is bolded in the original description at the top of this page.

Here's some examples of it in use:
- https://github.com/conda-forge/jupyter-cache-feedstock/pull/5
- https://github.com/conda-forge/sphinx-thebe-feedstock/pull/5

[the remark this answers is not in the document]
Comment (mmcky) [reply .github/CODEOWNERS:1]: thanks -- only checking as the `.github/CODEOWNERS` had a message at the top saying it didn't exist -- when I thought it did already. I'll merge this now.

**Pull request merged**